### PR TITLE
PnP Samples Cleanup and Refactorin

### PR DESCRIPTION
**Pull request by barustum**
- Clean up PnP samples
- Change PnPServiceSamples to follow same format as PnPServiceSamples

### CODE REVIEW

File: iot-hub/Samples/device/PnpDeviceSamples/TemperatureController/TemperatureControllerSample.cs [hunk 1]
```diff
@@ -244,80 +255,84 @@ private async Task TargetTemperatureUpdateCallbackAsync(TwinCollection desiredPr
...
+                    $"received an update which is not associated with a valid property.\n{desiredProperties.ToJson()}");
             }
         }
 
         // Report the property updates on "deviceInformation" component.
         private async Task UpdateDeviceInformationAsync(CancellationToken cancellationToken)
         {
-            string componentName = "deviceInformation";
+            const string componentName = "deviceInformation";
+            string messageHeader = $"Property: Update - component = \"{componentName}\"";
```
Comment (abhipsaMisra) [inline iot-hub/Samples/device/PnpDeviceSamples/TemperatureController/TemperatureControllerSample.cs:272]: nice!

File: iot-hub/Samples/service/DigitalTwinClientSamples/TemperatureController/TemperatureControllerSample.cs [hunk 2]
```diff
@@ -15,6 +14,11 @@ namespace Microsoft.Azure.Devices.Samples
 {
     public class TemperatureControllerSample
     {
+        private const string Thermostat1Component = "thermostat1";
+
+        private static readonly Uri DeviceTemperatureControllerSampleUri =
```
Comment (abhipsaMisra) [inline iot-hub/Samples/service/DigitalTwinClientSamples/TemperatureController/TemperatureControllerSample.cs:19]: why this?

File: iot-hub/Samples/service/DigitalTwinClientSamples/TemperatureController/TemperatureControllerSample.cs [hunk 3]
```diff
@@ -59,77 +65,83 @@ private async Task UpdateDigitalTwinComponentPropertyAsync()
...
-            HttpOperationHeaderResponse<DigitalTwinUpdateHeaders> updateDigitalTwinResponse = await _digitalTwinClient.UpdateDigitalTwinAsync(_digitalTwinId, updateOperation.Serialize());
+            _logger.LogDebug($"Update the {targetTemperaturePropertyName} property under component {Thermostat1Component} on the {_digitalTwinId} `" +
+                $"digital twin to {desiredTargetTemperature}.");
+            HttpOperationHeaderResponse<DigitalTwinUpdateHeaders> updateDigitalTwinResponse = await _digitalTwinClient
+                .UpdateDigitalTwinAsync(_digitalTwinId, updateOperation.Serialize());
 
             _logger.LogDebug($"Update {_digitalTwinId} digital twin response: {updateDigitalTwinResponse.Response.StatusCode}.");
 
+            // Amount of seconds to wait after updating targetTemperature property in order to allow the device to simulate temperature change
+            const int delay = 15;
```
Comment (abhipsaMisra) [inline iot-hub/Samples/service/DigitalTwinClientSamples/TemperatureController/TemperatureControllerSample.cs:115]: we do need this in a sample? I'd rather this be small, like 2-5secs or something. Else the sample will simply sit around for nothing.

Comment (abhipsaMisra) [reply iot-hub/Samples/service/DigitalTwinClientSamples/TemperatureController/TemperatureControllerSample.cs:115]: an unresponsive sample would not be the best user experience

File: iot-hub/Samples/service/DigitalTwinClientSamples/TemperatureController/TemperatureControllerSample.cs [hunk 4]
```diff
@@ -59,77 +65,83 @@ private async Task UpdateDigitalTwinComponentPropertyAsync()
...
-                    $"\nDevice returned status: {invokeCommandResponse.Body.Status}. \nReport: {invokeCommandResponse.Body.Payload}");
+                    $"\nDevice returned status: {invokeCommandResponse.Body.Status}.");
             }
             catch (HttpOperationException e)
             {
                 if (e.Response.StatusCode == HttpStatusCode.NotFound)
                 {
                     _logger.LogWarning($"Unable to execute command {rebootCommandName} on {_digitalTwinId}." +
-                        $"\nMake sure that the device sample TemperatureController located in https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/feature/digitaltwin/iot-hub/Samples/device/PnpDeviceSamples/TemperatureController is also running.");
+                        $"\nMake sure that the device sample TemperatureController located in {DeviceTemperatureControllerSampleUri.AbsoluteUri} is also running.");
```
Comment (abhipsaMisra) [inline iot-hub/Samples/service/DigitalTwinClientSamples/TemperatureController/TemperatureControllerSample.cs:144]: why not store it in a const string directly, any advantage of putting it into a "Uri"? We wouldn't be using the value for anything other than printing.

File: iot-hub/Samples/service/DigitalTwinClientSamples/Thermostat/ThermostatSample.cs [hunk 5]
```diff
@@ -57,35 +59,44 @@ private async Task UpdateTargetTemperaturePropertyAsync()
         {
             const string targetTemperaturePropertyName = "targetTemperature";
             var updateOperation = new UpdateOperationsUtility();
-
+
```
Comment (abhipsaMisra) [inline iot-hub/Samples/service/DigitalTwinClientSamples/Thermostat/ThermostatSample.cs:62]: nit: extra space

File: iot-hub/Samples/service/DigitalTwinClientSamples/TemperatureController/TemperatureControllerSample.cs (hunk 2 above)
Comment (barustum) [reply iot-hub/Samples/service/DigitalTwinClientSamples/TemperatureController/TemperatureControllerSample.cs:19]: This Uri is used in multiple methods in this file:
InvokeRebootCommandAsync
InvokeGetMaxMinReportCommandAsync

It's being used to warn the user to run the device sample before running the service sample. Created this field here to ensure that if the location of the url changed for instance, it's only updated once.

Comment (barustum) [reply iot-hub/Samples/service/DigitalTwinClientSamples/TemperatureController/TemperatureControllerSample.cs:19]: Will change to string. Also renaming.

File: iot-hub/Samples/service/PnpServiceSamples/TemperatureController/TemperatureControllerSample.cs [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+            _logger.LogDebug($"{_digitalTwinId} twin: \n{JsonConvert.SerializeObject(twin, Formatting.Indented)}");
+
+            return twin;
+        }
+
+        private async Task InvokeGetMaxMinReportCommandAsync()
+        {
+            const string getMaxMinReportCommandName = "getMaxMinReport";
+
+            // Create command name to invoke for component
```
Comment (barustum) [inline iot-hub/Samples/service/PnpServiceSamples/TemperatureController/TemperatureControllerSample.cs:64]: update comment to specify format for component level commands.

File: iot-hub/Samples/service/PnpServiceSamples/TemperatureController/TemperatureControllerSample.cs [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+        private async Task UpdateDigitalTwinComponentPropertyAsync()
+        {
+            // Choose a random value to assign to the targetTemperature property in thermostat1 component
+            int desiredTargetTemperature = Random.Next(0, 100);
+
+            const string targetTemperaturePropertyName = "targetTemperature";
+            string propertyUpdate = CreatePropertyPatch(targetTemperaturePropertyName,
+                JsonConvert.SerializeObject(desiredTargetTemperature),
+                Thermostat1Component);
+            string twinPatch = $"{{ \"properties\": {{\"desired\": {propertyUpdate} }} }}";
```
Comment (barustum) [inline iot-hub/Samples/service/PnpServiceSamples/TemperatureController/TemperatureControllerSample.cs:124]: nit: investigate changing from string to json object (twinCollection)

File: iot-hub/Samples/service/PnpServiceSamples/TemperatureController/TemperatureControllerSample.cs [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+
+            Twin twin = await _registryManager.GetTwinAsync(_digitalTwinId);
+
+            _logger.LogDebug($"Update the {targetTemperaturePropertyName} property under component {Thermostat1Component} on the {_digitalTwinId} " +
+                $"digital twin to {desiredTargetTemperature}.");
+            await _registryManager.UpdateTwinAsync(_digitalTwinId, twinPatch, twin.ETag);
+
+            // Amount of seconds to wait after updating targetTemperature property in order to allow the device to simulate temperature change
+            const int delay = 15;
```
Comment (barustum) [inline iot-hub/Samples/service/PnpServiceSamples/TemperatureController/TemperatureControllerSample.cs:133]: remove

File: iot-hub/Samples/service/PnpServiceSamples/Thermostat/ThermostatSample.cs [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+        {
+            _serviceClient = serviceClient ?? throw new ArgumentNullException(nameof(serviceClient));
+            _digitalTwinId = digitalTwinId ?? throw new ArgumentNullException(nameof(digitalTwinId));
+            _registryManager = registryManager ?? throw new ArgumentNullException(nameof(registryManager));
+            _logger = logger ?? LoggerFactory.Create(builder => builder.AddConsole()).CreateLogger<ThermostatSample>();
+        }
+
+        public async Task RunSampleAsync()
+        {
+            // Get and print the digital twin. If you do not know the exact structure of your digital twin, you can
```
Comment (barustum) [inline iot-hub/Samples/service/PnpServiceSamples/Thermostat/ThermostatSample.cs:31]: Remove command and check on PnpServiceSamples/TemperatureController.cs

File: iot-hub/Samples/service/PnpServiceSamples/Thermostat/ThermostatSample.cs [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+            var twinPatch = new Twin();
+            twinPatch.Properties.Desired[targetTemperaturePropertyName] = desiredTargetTemperature;
+
+            _logger.LogDebug($"Update the {targetTemperaturePropertyName} property on the " +
+                $"{_digitalTwinId} digital twin to {desiredTargetTemperature}.");
+
+            await _registryManager.UpdateTwinAsync(_digitalTwinId, twinPatch, twin.ETag);
+
+            // Amount of seconds to wait after updating targetTemperature property in order to allow the device to simulate temperature change
+            const int delay = 15;
```
Comment (barustum) [inline iot-hub/Samples/service/PnpServiceSamples/Thermostat/ThermostatSample.cs:71]: remove

**Pull request merged**